src/services/callHierarchy.ts

```diff
@@ -262,6 +262,12 @@ namespace ts.CallHierarchy {
                     }
                 }
             }

```

Comment:
Does the `else` make an actual difference here? Can this be reduced to the contained `if` check alone (or at least an `else if`)?

Reply:
It is OK to take the if clause out. But, not pretty sure, I think it split two conditions -- declaration and call expression.
```
function foo() {
    bar();
}

const bar = function () {    //declaration 
    baz();
}

function baz() {
}

bar()  // call expression
```
if followingSymbol is false and return, it means declaration, if followingSymbol is true, it means call expression.

I would have a close look at this when I have time.

Reply:
I test some cases, and think followingSymbol does mean whether is declaration or call expression. But some code are used for both condition to find returned value.
At least for the code I added, it is only used for the call expression of const function expression.
Do I still need to remove the else clause?

Reply:
This comment might have shown up in the wrong place. You have an `else` on [line 265](https://github.com/microsoft/TypeScript/pull/39579/files#diff-46de2fd0324aef6e77de5b5b11a8239cR265) immediately followed by an `if`. Could you not do an `else if` here?